recipes/minizip/all/conanfile.py

```diff
@@ -0,0 +1,138 @@
+from conans import ConanFile, CMake, tools
+import os
+
+
+class MinizipConan(ConanFile):
+    name = "minizip"
+    description = "minizip is a zip manipulation library written in C "
+    "that is supported on Windows, macOS, and Linux."
+    topics = ("conan", "minizip", "compression")
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/nmoinvaz/minizip"
+    license = "Zlib"
+    exports_sources = ["CMakeLists.txt", "patches/*"]
+    generators = "cmake_find_package"
+
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "compat": [True, False],
+        "zlib": [True, False],
+        "bzip2": [True, False],
+        "zstd": [True, False],
+        "lzma": [True, False],
+        "pkcrypt": [True, False],
+        "wzaes": [True, False],
+        "libcomp": [True, False],
+        "openssl": [True, False],
+        "libbsd": [True, False],
+        "brg": [True, False],
+        "signing": [True, False],
+        "compress_only": [True, False],
+        "decompress_only": [True, False]
```

Comment:
Can't these conflict?
How about:
```suggestion
        "compress": [True, False],
        "decompress": [True, False]
```
Set their default to `True`, and throw a `ConanInvalidConfiguration` if both are false.

Reply:
I had a look at the CMakeLists.txt and it turns out that minizip uses MZ_COMPRESS_ONLY and MZ_DECOMPRESS_ONLY to add the definitions MZ_ZIP_NO_DECOMPRESSION / MZ_ZIP_NO_COMPRESSION [https://github.com/nmoinvaz/minizip/blob/3ecff07e3cf76b848b9d6b219302c1c5311aa281/CMakeLists.txt#L395](url).

So the variable names are misleading and do not necessarily conflict.
For example mz_zip.c has a STORE compression method [https://github.com/nmoinvaz/minizip/blob/3ecff07e3cf76b848b9d6b219302c1c5311aa281/mz_zip.c#L1614](url)
I guess minizip without compression and decompression might be interesting if you just need to package files, especially already compressed ones like images.
Also those variables disable encryption/decryption functionality. See [https://github.com/nmoinvaz/minizip/blob/3ecff07e3cf76b848b9d6b219302c1c5311aa281/mz_strm_pkcrypt.c#L141](url) and [https://github.com/nmoinvaz/minizip/blob/3ecff07e3cf76b848b9d6b219302c1c5311aa281/mz_strm_pkcrypt.c#L163](url).

I would rename the options to no_compression and no_decompression but allow both to be set.

Reply:
Double negation `no_compress=False` is not clear imho.
Using no_compression and no_decompression would allow to set both options to True, which then should throw a ConanInvalidConfiguration.
How about merging them to one option?
e.g. `support: ["compress", "decompress", "both"]`
In this way, there will be no *impossible combinations*.

Reply:
I implemented your first suggestion in c9ae630, by enabling compression/decompression by default.
I do not consider both being turned of as invalid.

Reply:
What is the actual behavior when both are disabled?
So the library builds, but is non-functional? (only has a minizip_get_version() or similar)

Reply:
71a3fc9 adds a test case where a textfile is stored uncompressed in a zip archive by using the MZ_COMPRESS_METHOD_STORE instead of for example MZ_COMPRESS_METHOD_DEFLATE. This test succeeds even if compression and decompression is disabled.

Reply:
That does not really make sense, unless minizip does not really compress and only copies the input verbatim to the output.
Maybe I am asking too much and this is out of scope for a recipe, but some prints of the sizes of the (intermediate) buffers would clarify a lot.

Reply:
According to [4.1.7](https://pkware.cachefly.net/webdocs/APPNOTE/APPNOTE-6.3.9.TXT) the zip archive allows storing files uncompressed, even independent from all other files. So its kind of required that minizip implements this feature.